woocommerce/woocommerce-android
# Issue/873 clear backstack when site changed

Fixes #873 - After trying a variety of ways to reliably capture when the site has changed and clear all child fragments, I took a step back and tried a different approach. Instead of clearing fragments simply restart the main activity when the site has changed, and instead of relying on an `EventBus` event to tell us when the site has changed simply use `onActivityResult` to detect it.

To test:
* Open order detail
* Go to settings
* Change sites
* Return to the main activity

When you return to the main activity, no child fragments should be showing and it should refresh to show the new site's data. I recommend following the above steps both with and without "Don't keep activities" enabled.
